### DISCUSSION TITLE
v7.3.7 Release of PowerShell

### DISCUSSION THREAD

**Discussion by daxian-dbw** [PowerShell/PowerShell#20303] [Show and tell]
### Build and Packaging Improvements

<details>

<summary>

<p>Bump .NET SDK version to 7.0.401</p>

</summary>

<ul>
<li>Update 'ThirdPartyNotices.txt' (Internal 27602)</li>
<li>Update to use .NET SDK 7.0.401 (Internal 27591)</li>
<li>Remove <code>HostArchitecture</code> dynamic parameter for <code>osxpkg</code> (#19917)</li>
<li>Remove spelling CI in favor of GitHub Action (#20248)</li>
<li>Enable vPack provenance data (#20253)</li>
<li>Start using new packages.microsoft.com cli (#20252)</li>
<li>Add mariner arm64 to PMC release (#20251)</li>
<li>Add mariner arm64 package build to release build (#20250)</li>
<li>Make PR creation tool use <code>--web</code> because it is more reliable (#20247)</li>
<li>Update variable used to bypass the blocking check for multiple NuGet feeds (#20246)</li>
<li>Publish rpm package for rhel9 (#20245)</li>
<li>Add runtime and packaging type info for mariner2 arm64 (#20244)</li>
</ul>

</details>

### Documentation and Help Content

- Update man page to match current help for pwsh (#20249)

[7.3.7]: https://github.com/PowerShell/PowerShell/compare/v7.3.6...v7.3.7

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - DD754A80BF20E1F9E92AAAA38C70AA9483C4713D8B17E5801E3F0B12530B5416
- powershell_7.3.7-1.deb_amd64.deb
  - BC7806942A6F28060686621BD922DF555E61E11B00D5205896BABA251FFA5B1F
- powershell-7.3.7-1.cm.aarch64.rpm
  - 59DD614E2DD5D7703914431AC8ECD9254B748ECDDA81115AF7F08CC84E478862
- powershell-7.3.7-1.cm.x86_64.rpm
  - 7D9E879DFE6B1575041ED3917E3D395ABDC3F958D6E9C19FB55B705A7B3D4B8B
- powershell-7.3.7-1.rh.x86_64.rpm
  - 2A348CD6D01958781EE167D5E4C99981A75DF638EFCCDC142EC7E72A515FEF7C
- powershell-7.3.7-linux-alpine-x64.tar.gz
  - 5EEEDF5F7DA8E62F609D98BB4DA0FBD6DDD0BB8E7265A4AC9810915EB94E7FD3
- powershell-7.3.7-linux-arm32.tar.gz
  - F95EDB917661F91AFBAA968D37EC3C166674F826017D680F954F047CA400B6DA
- powershell-7.3.7-linux-arm64.tar.gz
  - 19A02EDE70F4C51AAA1342E6ED9E436BA61440688215CE711EE2580CB2B24867
- powershell-7.3.7-linux-x64.tar.gz
  - 18AB001FE03CF7F3357F1BF0E02E326FBF884DC7C3E98E0E89C6F52BC02CC302
- powershell-7.3.7-linux-x64-fxdependent.tar.gz
  - A94F24BE393D02DDF73863734FDB82213367227A09D3C6BF0049953153CF8838
- powershell-7.3.7-osx-arm64.pkg
  - 7AE91B9A86588F4E1B9199595169184B27F15C81D67754FF84C7B77D455ADF8E
- powershell-7.3.7-osx-arm64.tar.gz
  - 29206C630DFAF5F5114A6A03FD8C999BD08B1086E10D1D76991A75C4B089E167
- powershell-7.3.7-osx-x64.pkg
  - 4E25A0F564C46AFC605E6DA93B1ED165F18D7D60D3506971CA9701CB5F1A1588
- powershell-7.3.7-osx-x64.tar.gz
  - FBA732E0F2E9B77651EEE8B71FDAC08370B7F6456BC8FB6A1391CACCCA416B6E
- PowerShell-7.3.7-win-arm32.zip
  - 39942D24B839546ED6B073ACE5048E50F1555C8C610B75173D0E2E8CC920EA5B
- PowerShell-7.3.7-win-arm64.zip
  - 047DE9C873626A8833CB44833D6093305758853FDD28E77AA7D66D0BA78FACEB
- PowerShell-7.3.7-win-fxdependent.zip
  - C28A7C67DC7E4F88A375A632F98C2A3EB0334BD69185518933CB4DA66488C7AC
- PowerShell-7.3.7-win-fxdependentWinDesktop.zip
  - B571E7A5FC6BC1489E94B1AB566D8950CCB3EFB5EADBAE93ADF6248CFC0DAF44
- PowerShell-7.3.7-win-x64.msi
  - 9168BB9F166FF41A3E6A580DCB4189174F0563EE9DC438408C090A92FF1DCBFC
- PowerShell-7.3.7-win-x64.zip
  - 1C2988794A454C01990BCA872BB208590EC792189C115DBE0D71880EBCF760A6
- PowerShell-7.3.7-win-x86.msi
  - 71B71AA38DEB263F3776B480D3B6FE01850400A0E1FAA424BD47F9A72B43CD98
- PowerShell-7.3.7-win-x86.zip
  - 7FC7752C506F8792AA0139667A9F324B99E091FEB96DC5B90C79CD5BA2AFA981

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.3.7'>v7.3.7 Release of PowerShell</a>.</em>

**Comment by matracey** [comment]
@daxian-dbw Is `pwsh` no longer being distributed in an `msixbundle`? Was present in 7.3.6 but is missing here.

**Reply by HotCakeX** [reply]
I was thinking about the same thing

**Reply by gagneet** [reply]
It does seem to be present in the list of packages: 
![image](https://github.com/PowerShell/PowerShell/assets/203433/0a2ca1b2-25df-4092-ac14-877b0b26951a)

**Reply by vigneshwalker7** [reply]
great

**Comment by schonies** [comment]
Hi
Newbie here ...can someone explain what I'm looking at? Followed a link here to upgrade to the latest PowerShell

**Reply by FranLegon** [reply]
To update PowerShell Core, go to the "Assets" section of this release and choose what installer you want (probably PowerShell-7.3.7-win-x64.msi if you're on windows). Then just download it and run it.

**Reply by toocomputer** [reply]
@mitachnet i'm _not_ a newbie and find it frustrating, to say the least, to get a message that says "upgrade now", but with no instructions or direction whatsoever on where or how to do so. in reference to the message, if i don't want to check out the release page then what _do_ i do to upgrade (rhetorical)? 
![ps](https://github.com/PowerShell/PowerShell/assets/2061889/3e6ea30f-b548-43c3-a377-45e3552a660f)

**Reply by gagneet** [reply]
@bozoputer that link given in the message should lead you directly to the download page of the latest release. Is that not what you are expecting? Or are you wanting to see some release notes, etc.?

Yes, they have not added the "Upgrade" command line instruction, but if you are not a newbie, then you should know what that is and can run the upgrade from the PowerShell CLI only. (i.e. `winget install --id Microsoft.Powershell --source winget`)

**Reply by toocomputer** [reply]
@gagneet i meant newbie broadly speaking. but what i am or am not is irrelevant. is it too much to ask for the command you listed to be provided by the prompt?

i stand by my assertion that exhorting the user to take an action but not providing instructions or direction on how to do so is a poor user experience. the prompt is basically saying "upgrade (only if you already know how to) or go read some release notes and figure out what to do from there".

many command line utilities do a much better job of this (e.g., `[oh-my-zsh] Would you like to update? [Y/n]`).

**Reply by Michael-Steshenko** [reply]
I agree, a command in the prompt to upgrade would be nice.
I thought you couldn't update through the command line because I expected that if it were possible it would show the command in the prompt, so I've been using the installer.

**Reply by imatthewtanner** [reply]
@bozoputer I guess I've just gotten use to knowing how to upgrade. But now that you mentioned it I guess I can see your point. A command in the prompt would be nice.

**Reply by toocomputer** [reply]
@imatthewtanner dig. you’re sort of proving my point then, that the decision to elide the command is intentionally catering to seasoned PowerShell users such as yourself or @gagneet, or it’s an oversight.

either way, it creates a poor user experience, as it leaves users, such as myself, who haven’t programmed on a windows machine in years (and when i did i _intentionally_ stayed away from PowerShell) or users new to programming overall in the dark. 

they should do one or the other: list the command or remove the 
language inducing the user to complete an action.

we’re programmers—we know (or should know) how to duck duck go for answers, and i would’ve done that from the start (or just clicked the link) had the phrase “Upgrade now” not been used.

but since i know how to read and comprehend, that phrasing led me to believe there was an action i could take right there in the shell, and apparently there is, but how would i know what that action is unless the prompt lists it?

oh, i should _already_ know it (through osmosis, i guess). nice.

**Comment by vky5677** [comment]
What is this update is regarding? New features added or simply bug fixing.

**Reply by gagneet** [reply]
On the top of the page, click the arrow under "Build and Packaging Improvements", if it is not already open. Main enhancement seems to be "Bump .NET SDK version to 7.0.401"

**Comment by schoenfeldavi** [comment]
@FranLegon @vky5677  thanks!
Starting my journey into programming and I'm trying to ask questions that I would normally feel stupid asking. I don't want to be slowed down by previous handicaps.....

**Reply by FranLegon** [reply]
That's the way! Don't be afraid to ask. Also make heavy use of AI chats, they are great for learning.

**Reply by imatthewtanner** [reply]
@schoenfeldavi Same here! How's the saying go? "The only stupid question is the one not asked."

**Comment by venokk** [comment]
Ok!
